internal/controller/fluxinstance_controller.go

```diff
@@ -622,3 +640,34 @@ func (r *FluxInstanceReconciler) recordMetrics(obj *fluxcdv1.FluxInstance) error
 	reporter.RecordMetrics(unstructured.Unstructured{Object: rawMap})
 	return nil
 }
+
+func (r *FluxInstanceReconciler) notify(ctx context.Context, obj *fluxcdv1.FluxInstance, reason, eventType, msg string) {
+	notificationAddress := ""
+	if os.Getenv("NOTIFICATIONS_DISABLED") == "" && builder.ContainElementString(obj.GetComponents(), builder.MakeDefaultOptions().NotificationController) {
+		notificationAddress = fmt.Sprintf("http://%s.%s.svc.%s/",
+			builder.MakeDefaultOptions().NotificationController,
+			obj.GetNamespace(),
+			obj.GetCluster().Domain,
+		)
+	}
+
+	log := ctrl.LoggerFrom(ctx)
+
+	eventRecorder, err := events.NewRecorderForScheme(
+		r.Scheme,
+		r.EventRecorder,
+		log,
+		notificationAddress,
+		r.StatusManager)
+	if err != nil {
+		log.Error(err, "failed to create event recorder")
+		return
+	}
+
+	annotations := map[string]string{}
+	if obj.Status.LastAttemptedRevision != "" {
+		annotations[fluxcdv1.RevisionAnnotation] = obj.Status.LastAttemptedRevision
+	}
```

Comment:
Since in this case we are losing an event it'd be really nice to enhance the context of this log with the context of the lost event.

```suggestion
	annotations := map[string]string{}
	if obj.Status.LastAttemptedRevision != "" {
		annotations[fluxcdv1.RevisionAnnotation] = obj.Status.LastAttemptedRevision
	}

	eventRecorder, err := events.NewRecorderForScheme(
		r.Scheme,
		r.EventRecorder,
		log,
		notificationAddress,
		r.StatusManager)
	if err != nil {
		log.Error(err, "failed to create event recorder for event", "event", map[string]any{
			"instance":    map[string]any{"name": obj.Name, "namespace": obj.Namespace},
			"type":        eventType,
			"reason":      reason,
			"message":     msg,
			"annotations": annotations})
		return
	}
```

Reply:
The messages are already logged by the operator at each step. This would just duplicate tones of text in the controller logs.